### PR TITLE
GitAsyncRequest for profile

### PR DESCRIPTION
**Pull request by Riceqrisp**
Please review AsyncRequestGit.cs :)

### CODE REVIEW

File: GiTinder/Controllers/GitAsyncRequest.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+namespace GiTinder.Controllers
+{
+    [Route("test")]
+    [ApiController]
+    public class GitAsyncRequest
+    {
+        private readonly GiTinderContext _context;
+
+      
+        [HttpGet("username")]
```
Comment (adamgyulavari) [inline GiTinder/Controllers/GitAsyncRequest.cs]: @Riceqrisp this endpoint path should have a variable or a parameter for the username

**Review by adamgyulavari — CHANGES_REQUESTED** [review]
change the base of the pull request to week-01

File: GiTinder/Controllers/GitAsyncRequest.cs (hunk 1 above)
Comment (Riceqrisp) [reply GiTinder/Controllers/GitAsyncRequest.cs]: Created username parameter

**Comment by adamgyulavari** [comment]
@Riceqrisp change the base of the pull request to week-01

File: GiTinder/Controllers/GitAsyncRequest.cs (hunk 1 above)
Comment (Riceqrisp) [reply GiTinder/Controllers/GitAsyncRequest.cs]: changed the base

File: GiTinder/Controllers/GitAsyncRequest.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        //dev/user_profile/{username}
+        //async method is used everytime something should take long time like for requesting for data await allows certain line to run independently.
+        //if user doesn't exist in github then it shouldnt respond
+        public async Task<User> GetUserAsync(string username)
+        {
+            HttpClient client = new HttpClient();
+            //client.DefaultRequestHeaders.Clear() to clear headers being sure that it should only have requested headers. Learning
+            //
+            //Gihub API needs to track requests through header User Agent (https://developer.github.com/v3/#user-agent-required).
+            client.DefaultRequestHeaders.Add("User-Agent", "frog");
```
Comment (adamgyulavari) [inline GiTinder/Controllers/GitAsyncRequest.cs]: @Riceqrisp let's use a proper user agent name instead

File: GiTinder/Controllers/GitAsyncRequest.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        //    client.DefaultRequestHeaders.Add("User-Agent", "repofrog");
+        //    UserRepos newRepo = null;
+        //    HttpResponseMessage responseUserRepos = await client.GetAsync("https://api.github.com/users/Riceqrisp/repos");
+        //    if (responseUserRepos.IsSuccessStatusCode)
+        //    {
+        //        List<UserRepos> repos = await responseUserRepos.Content.ReadAsAsync<UserRepos>();
+        //    }
+            
+        //    return newRepo;
+        //}
```
Comment (adamgyulavari) [inline GiTinder/Controllers/GitAsyncRequest.cs]: @Riceqrisp don't leave outcommented code here

File: GiTinder/Services/User.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+//            {
+//                HttpClient client = new HttpClient();
+//                client.DefaultRequestHeaders.Add("User-agent", "GiTinderApp");
+//                HttpResponseMessage response = await client.GetAsync("https://api.github.com/repos/Riceqrisp/Career-pivot");
+
+//            }
+//        }
+//    }
+//}
```
Comment (adamgyulavari) [inline GiTinder/Services/User.cs]: @Riceqrisp remove the file

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: GiTinder/Controllers/GitAsyncRequest.cs (hunk 3 above)
Comment (Riceqrisp) [reply GiTinder/Controllers/GitAsyncRequest.cs]: Fixed :)

File: GiTinder/Services/User.cs (hunk 4 above)
Comment (Riceqrisp) [reply GiTinder/Services/User.cs]: Removed:)

File: GiTinder/Controllers/ValuesController.cs [hunk 5]
```diff
@@ -1,25 +1,36 @@
 ﻿using Microsoft.AspNetCore.Mvc;
 using System.Collections.Generic;
+<<<<<<< HEAD
+using System.Linq;
+using System.Threading.Tasks;
+using GiTinder.Data;
+using GiTinder.Services;
+using Microsoft.AspNetCore.Mvc;
+=======
+>>>>>>> ba9123ab47868405d039c3851fbcfee816e84dc4
```
Comment (adamgyulavari) [inline GiTinder/Controllers/ValuesController.cs]: @Riceqrisp remove conflicts

File: GiTinder/Migrations/GiTinderContextModelSnapshot.cs [hunk 6]
```diff
@@ -17,15 +16,20 @@ protected override void BuildModel(ModelBuilder modelBuilder)
 
             modelBuilder.Entity("GiTinder.Models.User", b =>
                 {
-                    b.Property<string>("UserName")
+                    b.Property<int>("Id")
```
Comment (adamgyulavari) [inline GiTinder/Migrations/GiTinderContextModelSnapshot.cs]: @Riceqrisp this should be already removed and username should be the key

File: GiTinder/Models/UserLanguages.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace GiTinder.Models
+{
+    public class UserLanguages
+    {
```
Comment (adamgyulavari) [inline GiTinder/Models/UserLanguages.cs]: @Riceqrisp this should have a single User and a single Language as property

File: GiTinder/Services/UserServices.cs [hunk 8]
```diff
@@ -1,9 +1,24 @@
...
 namespace GiTinder.Services
 {
+
+=======
+using System;
+using System.Linq;
+
+namespace GiTinder.Services
+{
+>>>>>>> ba9123ab47868405d039c3851fbcfee816e84dc4
```
Comment (adamgyulavari) [inline GiTinder/Services/UserServices.cs]: @Riceqrisp remove conflicts

File: GiTinder/Services/UserServices.cs [hunk 9]
```diff
@@ -12,6 +27,46 @@ public UserServices(GiTinderContext context)
         {
             _context = context;
         }
+<<<<<<< HEAD
+        //async method is used everytime something should take long time like for requesting for data await allows certain line to run independently.
+        //if user doesn't exist in github then it shouldnt respond
```
Comment (adamgyulavari) [inline GiTinder/Services/UserServices.cs]: @Riceqrisp do we really need the comments?

File: GiTinder/Services/UserServices.cs [hunk 10]
```diff
@@ -12,6 +27,46 @@ public UserServices(GiTinderContext context)
...
+        public async Task<User> GetUserAsync(string username)
```
Comment (adamgyulavari) [inline GiTinder/Services/UserServices.cs]: @Riceqrisp name this method like GetGithubProfileAsync

File: GiTinder/Services/UserServices.cs [hunk 11]
```diff
@@ -12,6 +27,46 @@ public UserServices(GiTinderContext context)
...
+        {
+    
+            HttpClient client = new HttpClient();
+            //client.DefaultRequestHeaders.Clear() to clear headers being sure that it should only have requested headers. Learning
+            //Gihub API needs to track requests through header User Agent (https://developer.github.com/v3/#user-agent-required).
+            client.DefaultRequestHeaders.Add("User-Agent", "GiTinderApp");
+            User rawUser = null;
+
+            HttpResponseMessage responseUser = await client.GetAsync("https://api.github.com/users/" + username);
```
Comment (adamgyulavari) [inline GiTinder/Services/UserServices.cs]: @Riceqrisp baseUrl should be a constant at the top

File: GiTinder/Services/UserServices.cs [hunk 12]
```diff
@@ -12,6 +27,46 @@ public UserServices(GiTinderContext context)
...
+<<<<<<< HEAD
+        //async method is used everytime something should take long time like for requesting for data await allows certain line to run independently.
+        //if user doesn't exist in github then it shouldnt respond
+        public async Task<User> GetUserAsync(string username)
+        {
+    
+            HttpClient client = new HttpClient();
+            //client.DefaultRequestHeaders.Clear() to clear headers being sure that it should only have requested headers. Learning
+            //Gihub API needs to track requests through header User Agent (https://developer.github.com/v3/#user-agent-required).
+            client.DefaultRequestHeaders.Add("User-Agent", "GiTinderApp");
```
Comment (adamgyulavari) [inline GiTinder/Services/UserServices.cs]: @Riceqrisp extract these two lines into a separate method

File: GiTinder/Services/UserServices.cs [hunk 13]
```diff
@@ -12,6 +27,46 @@ public UserServices(GiTinderContext context)
...
+                string Urls = null;
+                for (int i = 0; i < rawRepos.Count; i++)
+                {
+                   Urls = rawRepos[i].Url + " " + Urls;
+                }
+                _context.Users.Find(username).Repos = Urls;
+                _context.SaveChanges();
+            }
+            return rawRepos;
+=======
```
Comment (adamgyulavari) [inline GiTinder/Services/UserServices.cs]: @Riceqrisp remove conflict

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: GiTinder/Controllers/ValuesController.cs (hunk 5 above)
Comment (Riceqrisp) [reply GiTinder/Controllers/ValuesController.cs]: Values controller was removed by pulling week-01

File: GiTinder/Migrations/GiTinderContextModelSnapshot.cs (hunk 6 above)
Comment (Riceqrisp) [reply GiTinder/Migrations/GiTinderContextModelSnapshot.cs]: Id removed

File: GiTinder/Models/UserLanguages.cs (hunk 7 above)
Comment (Riceqrisp) [reply GiTinder/Models/UserLanguages.cs]: Created User property as a string.

File: GiTinder/Services/UserServices.cs (hunk 8 above)
Comment (Riceqrisp) [reply GiTinder/Services/UserServices.cs]: removed and changed namespace to GiTinder.Services.GitResponses

File: GiTinder/Services/UserServices.cs (hunk 9 above)
Comment (Riceqrisp) [reply GiTinder/Services/UserServices.cs]: Comments erased

File: GiTinder/Services/UserServices.cs (hunk 10 above)
Comment (Riceqrisp) [reply GiTinder/Services/UserServices.cs]: Re-named methods

File: GiTinder/Services/UserServices.cs (hunk 11 above)
Comment (Riceqrisp) [reply GiTinder/Services/UserServices.cs]: Changed to const string

File: GiTinder/Services/UserServices.cs (hunk 12 above)
Comment (Riceqrisp) [reply GiTinder/Services/UserServices.cs]: Extracted to HeadersSettingForGitHubApi()

Comment (Riceqrisp) [reply GiTinder/Services/UserServices.cs]: Moved private HttpClient client = new HttpClient(); to fields.

File: GiTinder/Services/UserServices.cs (hunk 13 above)
Comment (Riceqrisp) [reply GiTinder/Services/UserServices.cs]: Solved... hopefully.

**Pull request merged**